src/types.ts

```diff
@@ -1,3 +1,20 @@
+import { AppProps } from 'single-spa';
+import Aurelia, { IContainer, JitHtmlBrowserConfiguration } from 'aurelia';
+
+export type HostGetter = () => HTMLElement;
+
 export interface SingleSpaAureliaOptions {
-  bootstrap(configure: Function): Promise<void>;
+  // `component` should reference root component class, for instance `component: MyApp`.
+  component: unknown;
+  // Custom DI container that can be created via `DI.createContainer()` and
+  // passed custom dependencies (singleton classes, etc.).
+  container?: IContainer;
+  Aurelia: typeof Aurelia;
+  JitHtmlBrowserConfiguration: typeof JitHtmlBrowserConfiguration;
+  hostGetter?(): HTMLElement;
+  configure?(aurelia: Aurelia, props: AppProps): void;
```

Comment:
What do you envision this API being used for? Why would they need to change the Aurelia object based on props?

Reply:
I actually was wondering do we need to pass _props_ to the user in the same way as we do in _single-spa-angular_...

I would probably leave _configure()_ method but only with _aurelia_ argument since it's an instance and has _register()_ method for DI. The user might want to do something before mounting the app. *Might*. I don't predict anything. I will leave a decision for you in this case.

Reply:
The more common way within single-spa is to pass the props to the root component, instead of having to do the workaround with singleSpaPropsSubject. For example, in vue and react we just pass the single-spa props down to the root vue/react component as props.

Do aurelia components have the concept of props? If so let's just pass the props in that way.

Reply:
Shall we remove this now that the single-spa props are passed to the root component?